### ISSUE TITLE
[return-await] removes await for imports from mapped paths

### ISSUE THREAD

**Issue by bogdanb** [typescript-eslint/typescript-eslint#2606]
- [×] I have tried restarting my IDE and the issue persists.
- [×] I have updated to the latest version of the packages.
- [×] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/return-await": [
      "error",
      "always"
    ],
  }
}
```

```TS
// in file /src/main/assets/util/abort.ts:
export async function testReturnAwait(): Promise<string> {
  return "test";
}

// in file /src/main/assets/test/test.ts:
import { testReturnAwait } from "util/abort";

export async function errorExample(): Promise<string> {
  return await testReturnAwait(); // Error reported here, await removed on --fix
}
```
// My tsconfig:
```JSON
{
  "include": [
    "./.*.js",
    "./*.js",
    "./*.ts",
    "./src/main/assets/**/*.ts",
    "./src/main/assets/**/*.tsx",
    "./src/main/assets/**/*.js",
    "./src/main/assets/**/*.jsx"
  ],
  "compilerOptions": {
    "moduleResolution": "Node",
    "target": "es2020",
    "module": "es2020",
    "jsx": "react",
    "allowJs": true,
    "noEmit": true,
    "strict": true,
    "noImplicitAny": false,
    "noImplicitThis": true,
    "allowSyntheticDefaultImports": true,
    "experimentalDecorators": true,
    "baseUrl": ".",
    "paths": {
      "util/*": ["./src/main/assets/util/*"]
    }
  }
},
```

**Expected Result**

I would expect the rule to accept the code as given. If `errorExample()` did not have an `await`, I expected the rule should report an error, and propose adding the `await`.

**Actual Result**

The rule reports an error in `errorExample()`, at the line marked with the comment. If `--fix` is passed to `eslint`, the `await` is removed.

**Additional Info**

The invalid behavior above happens only when calling a function from a path-mapped import. (Note the end my tsconfig.) If I replace the import with `import { testReturnAwait } from "../util/abort"` then it works as expected.

It’s possible this is actually a special case of #2598: I suspect the path mapping is ignored/unsupported, and the function is actually typed as `any`.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.2.0` |
| `@typescript-eslint/parser`        | `4.2.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `14.7.0` |

**Comment by bradzacher** [comment]
> I suspect the path mapping is ignored/unsupported

We use TypeScript itself to do our parsing and type analysis, so nothing is unsupported.
You are correct though - generally if there is something showing up as `any`, it's because there's a config problem which is causing TS to not resolve the imports.

Could you please create an isolated repro repo to aid with investigation here?

**Comment by bogdanb** [comment]
I encountered another weird error when investigating this, it looks like a bug somewhere when using mapped paths on Windows.

I’ll keep investigating later, I have another priority at work right now.
